ksql-engine/src/main/java/io/confluent/ksql/util/QueryMetadata.java

```diff
@@ -141,7 +142,9 @@ public boolean hasEverBeenStarted() {
   }
 
   public void close() {
-    kafkaStreams.close();
+    int closeWaitDuration = Integer.parseInt(
+        streamsProperties.getOrDefault("close.wait", "60").toString());
```

Comment:
A bit of a guess on my part.  But I figured we'd want this value configurable.  Since this is an internal setting only, I think setting an internal config is acceptable.

Reply:
Where is `close.wait` configure defined, or are we practically always using 60 seconds?

Reply:
What factors influence how long the stream takes to close?  I'm thinking:
 * flushing of offsets and produced messages to Kafka: which can block past the close time if there are issues with Kafka
 * UDFs that take a long time, e.g, any calling another system.
 * Stream / Kafka client config that means huge volumes are being batch fetched

Anything else?

Of these, it seems the most likely is issues talking to other services. 

I guess it makes sense to make this configurable so that users can choose to wait longer for any in-flight messages to be processed and any results/offsets committed.   However, we generally use `KsqlConfig` for defining our Config.  It shouldn't be much more work to have `KsqlConfig` define a config for this, with a default, and pass in the `closeTimeout` in the `QueryMetadata` constructor.  (The `QueryExecutor` that creates both `PersistentQuery` and `TransientQuery`'s has a `ksqlConfig` field).

Reply:
Ack